### ISSUE TITLE
Can I use only part of the output projection during the inference?

### ISSUE THREAD

**Issue by SefaZeng** [NVIDIA/FasterTransformer#541]
I have a multilingual translation model and the vocab size is 250k. I found that the GPU usage of the faster-transformer is very high compared to deploying the server with PyTorch, I guess this is because the faster-transformer uses the output-projection of the entire vocab size for prediction, while the PyTorch version of the server can simply select only part of the vocab for prediction.
I wonder if such a modification of the faster-transformer is possible, since I'm not very familiar with cuda and cpp?

**Comment by byshiue** [comment]
Yes, you can pass same pointer of vocab weights in FT.

**Comment by SefaZeng** [comment]
> Yes, you can pass same pointer of vocab weights in FT.

Is it possible if I want to change the vocab size list every time I receive a new query? In pytorch, I can set the vocab like this:
```
index_lang_1 = [1,2,3]
index_lang_2 = [4,5,6]
# vocab weight's size is [6, 1024] which means 6 token in vocab
vocab.weight = vocab.weight[index_lang_1] if request_lang == "en" else vocab.weight[index_lang_2]
'''
Is this possible for faster-transformer to change the source code? This may be a silly question because I'm not very familiar with cuda and cpp... 
Thanks in advance!! :-）

**Comment by byshiue** [comment]
You need to change the current codes. You can pass the vocab pointer as an input, and use it to compute logits.
